saleor/order/utils.py

```diff
@@ -585,6 +598,11 @@ def change_order_line_quantity(
                 catalogue_discount, line, new_quantity, currency
             )
 
+        if is_line_level_voucher(order.voucher):
```

Comment:
`change_order_line_quantity` is called inside `add_variant_to_order`. This is called in the loop. It means that you're going to call `create_or_update_voucher_discount_objects_for_order` for each item in the loop. Is it expected? Shouldn't we call it only once at the end?